crates/cxx-qt-lib/src/gui/qpolygon.cpp

```diff
@@ -32,3 +32,20 @@ static_assert(!::std::is_trivially_copy_constructible<QPolygon>::value);
 static_assert(!::std::is_trivially_destructible<QPolygon>::value);
 
 static_assert(QTypeInfo<QPolygon>::isRelocatable);
+
+namespace rust {
+namespace cxxqtlib1 {
+const QVector<QPoint>&
+qpolygonAsQVectorQPointRef(const QPolygon& shape)
+{
+  return static_cast<const QVector<QPoint>&>(shape);
+}
+
+QVector<QPoint>&
+qpolygonAsQVectorQPointRef(QPolygon& shape)
+{
+  return static_cast<QVector<QPoint>&>(shape);
+}
```

Comment:
I appreciate the effort here to also allow upcasting the QPolygon to it's base class `QList<QPoint>`.

Note that we will replace this implementation with the generalized version that @BenFordTytherington is working one once it lands:
https://github.com/KDAB/cxx-qt/pull/1159
For now we can keep it as it should be compatible API-wise